dsl_parser/import_resolver/abstract_import_resolver.py

```diff
@@ -42,12 +47,32 @@ def fetch_import(self, import_url):
 
 
 def read_import(import_url):
-    try:
-        with contextlib.closing(urllib2.urlopen(import_url)) as f:
-            return f.read()
-    except Exception, ex:
-        ex = exceptions.DSLParsingLogicException(
-            13, 'Import failed: Unable to open import url '
-            '{0}; {1}'.format(import_url, str(ex)))
-        ex.failed_import = import_url
-        raise ex
+    if import_url.startswith('file://'):
+        try:
+            with contextlib.closing(urllib2.urlopen(import_url)) as f:
+                return f.read()
+        except Exception, ex:
+            ex = exceptions.DSLParsingLogicException(
+                13, 'Import failed: Unable to open import url '
+                    '{0}; {1}'.format(import_url, str(ex)))
+            raise ex
+    else:
+        try:
+            r = requests.get(import_url, timeout=DEFAULT_REQUEST_TIMEOUT)
+            num_retries = 0
+            while (not r.status_code == requests.codes.ok) and num_retries < DEFAULT_NUMBER_RETRIES:
```

Comment:
we discussed this a decided that request code should not be taken into consideration with regards to retrying